### ISSUE TITLE
[Due for payment 2025-02-13] [$250] mWeb/Chrome - Chat - Edition composer is cutted when editing multiline message near the top of screen

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#52138]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.58-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/5184687&group_by=cases:section_id&group_order=asc&group_id=292106
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Open the staging.new.expensify.com website
2. Open any chat with long history
3. Long tap on a multiline message displayed near the top of the screen
4. Select "Edit Message"
5. Verify that the chat is scrolled up and edition composer is fully visible

## Expected Result:
When trying to edit a multiline message displayed near the top of the screen, the chat should be scrolled up to make the composer fully visible

## Actual Result:
When trying to edit a multiline message displayed near the top of the screen, the chat is not scrolled up when the keyboard is opened and the edition composer gets cutted off by the chat header

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/b77258f0-424b-4807-88dd-153853c50266

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021857151760601711832</li>
        <li>Upwork Job ID: 1857151760601711832</li>
        <li>Last Price Increase: 2025-01-04</li>
<li>Automatic offers: </li>
<ul>
<li>ZhenjaHorbach | Contributor | 105195249</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @abekkala</details>

**Comment by abekkala** [comment]
confirmed the composer should not be partially hidden by the header

**Comment by wildan-m** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

The mWeb/Chrome Chat Edition composer is cut off when editing a multiline message near the top of the screen.

### What is the root cause of that problem?

The scroll pushed by soft input keyboard and in some cases it will cut the editor.

### What changes do you think we should make in order to solve the problem?

I think we need to scroll to the edited report action before we perform editing. This way we can ensure that the edited message is near the top of the soft input keyboard

In src/pages/home/report/ReportActionItem.tsx modify this use effect to add `scrollToIndex`
```tsx

    useEffect(() => {
        if (prevDraftMessage !== undefined || draftMessage === undefined) {
            return;
        }

        InteractionManager.runAfterInteractions(() => {
            requestAnimationFrame(() => {
                reportScrollManager.scrollToIndex(index);
            });
        })

        focusComposerWithDelay(textInputRef.current)(true);
    }, [prevDraftMessage, draftMessage]);


```

Optional:
Limit to `Browser.isMobileChrome` or `Browser.isMobile` if we don't want this behavior on other platforms

### What alternative solutions did you explore? (Optional)
N/A

**Comment by wildan-m** [comment]
[Proposal Updated](https://github.com/Expensify/App/issues/52138#issuecomment-2496554118)

- Add branch link

**Comment by abekkala** [comment]
@thesahindia can you take a look at the updated proposal from @wildan-m

**Comment by wildan-m** [comment]
[Proposal Updated](https://github.com/Expensify/App/issues/52138#issuecomment-2496554118)
- suggest disable animation

@thesahindia what about disabling the scroll animation?

**Comment by wildan-m** [comment]
[Proposal Updated](https://github.com/Expensify/App/issues/52138#issuecomment-2496554118)
- Add alternative proposal

> it doesn't look good.

@thesahindia if we check this PR https://github.com/Expensify/App/pull/2320 that behavior is actually implemented but only for native. I think we should also implement the behavior for mweb

**Comment by wildan-m** [comment]
@ZhenjaHorbach please also review my alternative proposal. I've noticed that pattern also already implemented in `onFocus` event but only for native. I assume that would be the accepted behavior?

**Comment by wildan-m** [comment]
@ZhenjaHorbach have a look at these native behavior: 

https://github.com/user-attachments/assets/229239e5-313b-4799-9b07-4db8ccd8ae5a

https://github.com/user-attachments/assets/867e4e8d-8005-4d0f-9d06-81bf0261185b

That is how current main looks like.

In my opinion, my proposed solution has similar behavior, and it's normal if mobile web feel slower than native.

**Comment by abekkala** [comment]
I'll check with CS and SWM after the new year

**Comment by abekkala** [comment]
https://expensify.slack.com/archives/C04878MDF34/p1736347115909849

**Comment by 289Adam289** [comment]
Hi! I am Adam from SWM and I can work on this issue.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by 289Adam289** [comment]
I was able to reproduce on emulator, but I had a problem reproducing on physical device. I will continue to work on this today.

https://github.com/user-attachments/assets/acb17a12-e021-47c5-a2cd-7e0eae860a33

**Comment by 289Adam289** [comment]
### Update
I've been addressing this issue and I've implemented one fix based on  @wildan-m proposal. Unfortunately it requires platform-specific changes  in `ReportActionItemMessadeEdit.tsx` file:
```
if (isMobileChrome() && reportScrollManager.ref?.current) {
      reportScrollManager.ref.current.scrollToIndex({index, animated: false});
}
```
I've been looking for code that will work on all platforms but due to how android keyboard works it is really problematic.
This is how it looks on android and iOS mWeb:


https://github.com/user-attachments/assets/36ae025d-2990-49eb-9f10-5a00165626f8 



https://github.com/user-attachments/assets/7b1e81a5-e0cd-4f2d-a11d-1db9e0e89a12


Let me know how should we proceed:
1. We can go with this solution
2. I can invest more time in finding platform-independent solution, which may prove very time consuming.
3. Or should I improve current implementation visuals
cc @ZhenjaHorbach @abekkala

**Comment by 289Adam289** [comment]
This might require more investigation due to difference between Android and iOS web keyboards. Simply using `scrollToIndex(index)` results in composer being hidden under the keyboard, even with `keyboardAvoidingView`.

**Comment by abekkala** [comment]
@francoisl there is some discussion on how to go about a fix for this one and I figured I should get an internal engineers opinion on this: 

https://github.com/Expensify/App/issues/52138#issuecomment-2596415615

**Comment by francoisl** [comment]
Sorry for the delay here, catching up on this.

>Let me know how should we proceed:
>
>1. We can go with this solution
>2. I can invest more time in finding platform-independent solution, which may prove very time consuming.
>3. Or should I improve current implementation


@289Adam289 I'll assign you so we can keep this moving. _Ideally_ we could find a way to improve your current solution to prevent the composer from jumping above the keyboard on Android mWeb, but I also don't think it's worth spending a ton of time on this - so if you can't figure out anything, let's go with your current proposal.

**Comment by francoisl** [comment]
Oh cool you were already assigned :D

**Comment by 289Adam289** [comment]
@francoisl Could you explain what do you mean by 

> jumping above the keyboard on Android mWeb

If i understand correctly this jump is caused by keyboard itself and can be observed with the main composer as well.

https://github.com/user-attachments/assets/12686927-947c-472d-baf2-009bef2b230a

**Comment by francoisl** [comment]
Yes that's the "jump" I was talking about, for lack of a better term.

Understood though, I thought somehow it was due to your changes. Feel free to submit a PR for the fix you have. Thanks!

**Comment by abekkala** [comment]
### PAYMENT SUMMARY FOR FEB 13

- Fix: @289Adam289 - no payment needed SWM
- PR Review: @ZhenjaHorbach [$250] [Offer](https://www.upwork.com/nx/wm/offer/105195249)

**Comment by abekkala** [comment]
@ZhenjaHorbach payment sent and contract ended - thank you! 🎉
